assets/javascripts/discourse/widgets/qa-comment.js

```diff
@@ -8,24 +8,86 @@ import { ajax } from "discourse/lib/ajax";
 import bootbox from "bootbox";
 import I18n from "I18n";
 
+createWidget("qa-comment-editor", {
+  tagName: "div",
+  buildKey: (attrs) => `qa-comment-editor-${attrs.id}`,
+
+  buildClasses(attrs) {
+    return ["qa-comment-editor", `qa-comment-editor-${attrs.id}`];
+  },
+
+  defaultState(attrs) {
+    return { updatingComment: false, value: attrs.raw };
+  },
+
+  html(attrs, state) {
+    return [
+      h("textarea", attrs.raw),
+      this.attach("button", {
+        action: "editComment",
+        disabled: state.updatingComment,
+        contents: I18n.t("qa.post.edit_comment"),
+        icon: "pencil-alt",
+        className: "btn-primary qa-comment-editor-submit",
+      }),
+      this.attach("link", {
+        action: "collapseEditor",
+        className: "qa-comment-editor-cancel",
+        contents: () => I18n.t("qa.post.cancel_comment"),
+      }),
+    ];
+  },
+
+  input(e) {
+    this.state.value = e.target.value;
+  },
+
+  keyDown(e) {
+    if (e.key === "Enter" && (e.ctrlKey || e.metaKey)) {
+      this.sendWidgetAction("editComment");
+    }
+  },
+
+  editComment() {
+    this.state.updatingComment = true;
+
+    return ajax("/qa/comments", {
+      type: "PUT",
+      data: {
+        comment_id: this.attrs.id,
+        raw: this.state.value,
+      },
+    })
+      .then((response) => {
+        this.sendWidgetAction("updateComment", response);
+      })
+      .catch(popupAjaxError)
+      .finally(() => {
+        this.sendWidgetAction("collapseEditor");
+        this.state.updatingComment = false;
```

Comment:
Btw if the request fails do you have to reset `this.state.raw`?

Reply:
Good point here. If the requests fails, I've opted to keep the composer open so we don't need to reset the state.